vsphere/stackstate_checks/vsphere/vsphere.py

```diff
@@ -989,22 +989,24 @@ def _vsphere_vms(self, content, domain="Unspecified", regexes=None, include_only
             labels = []
             if not self._is_excluded(c, regexes, include_only_marked):
                 hostname = c.name
-
-                if isinstance(c, vim.VirtualMachine):
-                    topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.VM
-                    topology_tags["name"] = c.name
-                    topology_tags["datastore"] = c.datastore[0]._moId
-                    topology_tags["layer"] = TOPOLOGY_LAYERS.VM
-                    topology_tags["domain"] = domain
-
-                    sts_identifiers, labels = self.extract_tags("VirtualMachine", c._moId)
-                    topology_tags["identifiers"] = sts_identifiers
-
-                    add_label_pair(labels, "name", topology_tags["name"])
-                    add_label_pair(labels, "guestId", c.config.guestId)
-                    add_label_pair(labels, "guestFullName", c.config.guestFullName)
-                    add_label_pair(labels, "numCPU", c.config.hardware.numCPU)
-                    add_label_pair(labels, "memoryMB", c.config.hardware.memoryMB)
+                try:
+                    if isinstance(c, vim.VirtualMachine):
+                        topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.VM
+                        topology_tags["name"] = c.name
+                        topology_tags["layer"] = TOPOLOGY_LAYERS.VM
+                        topology_tags["domain"] = domain
+
+                        sts_identifiers, labels = self.extract_tags("VirtualMachine", c._moId)
+                        topology_tags["identifiers"] = sts_identifiers
+
+                        topology_tags["datastore"] = c.datastore[0]._moId
```

Comment:
What happens if `c.datastore` is empty?

Reply:
@mpvvliet it will go in exception and continue the check with creating the component and there is no datastore associated with it.

Reply:
Have you considered rewriting it using a `for` loop rather than straight indexing element `0`? Then if it's empty it will at least continue and we will get further information as well.